source/extensions/common/dynamic_forward_proxy/dns_cache_impl.cc

```diff
@@ -32,6 +32,12 @@ DnsCacheImpl::DnsCacheImpl(
       max_hosts_(PROTOBUF_GET_WRAPPED_OR_DEFAULT(config, max_hosts, 1024)) {
   tls_slot_->set([](Event::Dispatcher&) { return std::make_shared<ThreadLocalHostInfo>(); });
   updateTlsHostsMap();
+
+  if (config.has_dns_cache_circuit_breaker()) {
+    auto cb_stats = generateDnsCacheCircuitBreakersStats(*scope_);
+    resource_manager_ = std::make_unique<DnsCacheResourceManager>(
```

Comment:
To simplify the logic, always make a resource manager, and just fill it with default values, per the other thread around what happens when there is no config.